formats/polus-czi-extract-plugin/src/main.py

```diff
@@ -1,18 +1,189 @@
+        czi = czifile.CziFile(file_path,detectmosaic=False)
+        subblocks = [s for s in czi.filtered_subblock_directory if s.mosaic_index is not None]
+        
+        ind = {'X': [],
+               'Y': [],
+               'Z': [],
+               'C': [],
+               'T': [],
+               'Row': [],
+               'Col': []}
+        
+        # Get the indices of each FOV
+        for s in subblocks:
+            scene = [dim.start for dim in s.dimension_entries if dim.dimension=='S']
+            if scene is not None and scene[0] != 0:
+                continue
+            
+            for dim in s.dimension_entries:
+                if dim.dimension=='X':
+                    ind['X'].append(dim.start)
+                elif dim.dimension=='Y':
+                    ind['Y'].append(dim.start)
+                elif dim.dimension=='Z':
+                    ind['Z'].append(dim.start)
+                elif dim.dimension=='C':
+                    ind['C'].append(dim.start)
+                elif dim.dimension=='T':
+                    ind['T'].append(dim.start)
+                    
+        row_conv = {y:row for (y,row) in zip(np.unique(np.sort(ind['Y'])),range(0,len(np.unique(ind['Y']))))}
+        col_conv = {x:col for (x,col) in zip(np.unique(np.sort(ind['X'])),range(0,len(np.unique(ind['X']))))}
+        
+        ind['Row'] = [row_conv[y] for y in ind['Y']]
+        ind['Col'] = [col_conv[x] for x in ind['X']]
+        
+        with BioReader(file_path) as br:
+            
+            metadata = br.metadata
+            chan_names = br.cnames
+        
+        for s,i in zip(subblocks,range(0,len(subblocks))):
+            
+            Z = None if len(ind['Z'])==0 else ind['Z'][i]
+            C = None if len(ind['C'])==0 else ind['C'][i]
+            T = None if len(ind['T'])==0 else ind['T'][i]
+        
+            out_file_path = out_path.joinpath(_get_image_name(base_name,
+                                                              row=ind['Row'][i],
+                                                              col=ind['Col'][i],
+                                                              Z=Z,
+                                                              C=C,
+                                                              T=T))
+            
+            dims = [_get_image_dim(s,'Y'),
+                    _get_image_dim(s,'X'),
+                    _get_image_dim(s,'Z'),
+                    _get_image_dim(s,'C'),
+                    _get_image_dim(s,'T')]
+            
+            data = s.data_segment().data().reshape(dims)
+            
+            write_thread(out_file_path,
+                         data,
+                         metadata,
+                         chan_names[C])
+
+def main(input_dir: Path,
+         output_dir: Path
+         ) -> None:
+
+    logger.info('Extracting tiffs and saving as ome.tif...')
+    files = [f for f in Path(input_dir).iterdir() if f.suffix=='.czi']
```

Comment:
In this case no. Since the input is `genericData` rather than `collection`, there should not be any subdirectories. We should only check for an `image` subdirectory when the input is `collection`.